src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -263,15 +264,20 @@ private static object[] BindArguments(Unpacker unpacker, IReadOnlyList<Type> par
 
         public void WriteMessage(HubMessage message, IBufferWriter<byte> output)
         {
-            using (var stream = new LimitArrayPoolWriteStream())
+            var writer = MemoryBufferWriter.Get();
```

Comment:
Does this write everything to array pool buffers just to get a length for prefix, then copies everything out of the buffers to the output after writing the prefix?

Reply:
Yep. We discussed ways to improve this but haven't enacted any concrete plan yet.